packages/dialog/index.tsx

```diff
@@ -0,0 +1,322 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as React from 'react';
+import classnames from 'classnames';
+
+// @ts-ignore no .d.ts file
+import {MDCDialogFoundation, MDCDialogAdapter, util} from '@material/dialog/dist/mdc.dialog';
+// @ts-ignore no .d.ts file
+import {ponyfill} from '@material/dom/dist/mdc.dom';
+/* eslint-disable no-unused-vars */
+import DialogContent, {DialogContentProps} from './DialogContent';
+import DialogFooter, {DialogFooterProps} from './DialogFooter';
+import DialogTitle, {DialogTitleProps} from './DialogTitle';
+/* eslint-enable no-unused-vars */
+import DialogButton from './DialogButton';
+import {cssClasses, LAYOUT_EVENTS} from './constants';
+import {FocusTrap} from 'focus-trap';
+
+export type ChildTypes<ChildProps> = DialogTitle<ChildProps> | DialogContent<ChildProps>| DialogFooter<ChildProps>;
+export type ChildProps<T> = DialogTitleProps<T> | DialogContentProps<T> | DialogFooterProps<T>;
+
+export interface DialogProps<T> extends React.HTMLProps<T> {
+  autoStackButtons?: boolean;
+  children?: ChildTypes<ChildProps<T>>[] | ChildTypes<ChildProps<T>>;
+  className?: string;
+  escapeKeyAction?: string;
+  id?: string;
+  onClose?: (action: string) => void;
+  onClosing?: (action: string) => void;
+  onOpen?: () => void;
+  onOpening?: () => void;
+  open?: boolean;
+  scrimClickAction?: string;
+  tag?: string;
+};
+
+interface DialogState {
+  classList: Set<string>;
+}
+
+function isDialogTitle(element: any): element is DialogTitle<any> {
+  return element.type === DialogTitle;
+}
+
+function isDialogContent(element: any): element is DialogContent<any> {
+  return element.type === DialogContent;
+}
+
+class Dialog<T extends {} = HTMLElement> extends React.Component<
+  DialogProps<T>,
+  DialogState
+  > {
+  focusTrap?: FocusTrap;
+  foundation: MDCDialogFoundation;
+  dialogElement: React.RefObject<HTMLElement> = React.createRef();
+  labelledBy?: string;
+  describedBy?: string;
+
+  static defaultProps: Partial<DialogProps<HTMLElement>> = {
+    autoStackButtons: true,
+    className: '',
+    onOpening: () => {},
+    onClosing: () => {},
+    tag: 'div',
+    id: 'mdc-dialog',
+    open: false,
```

Comment:
Not really, however, the reasoning for it might be trivial. When I leave these two off it tells me codecoverage is 100% for functions. Otherwise it goes down to 95.56%. However, I don't really understand why that is the case. Am I not testing it in this test `#adapter.notifyOpened calls props.onOpen'?